libs/platform/src/lib/table/domain/table-data-provider.ts

```diff
@@ -1,10 +1,343 @@
+     * @param adapter
+     * @returns whether or not item should be included in data source array.
+     */
+    protected filterDate<D = FdDate>(item: T, filter: CollectionDateFilter, adapter: DatetimeAdapter<D>): boolean {
+        if (!adapter && isDevMode()) {
+            console.error(
+                'In order to filter date columns, please provide DateTime adapter in your TableDataProvider constructor.'
+            );
+        }
+
+        const filterValue = filter.value;
+        const filterValue2 = filter.value2;
+        const itemValue = this._getNestedValue(filter.field, item);
+        const diff = adapter.compareDate(itemValue, filterValue);
+        let result = false;
+
+        switch (filter.strategy) {
+            case 'after':
+                result = diff > 0;
+                break;
+            case 'onOrAfter':
+                result = diff >= 0;
+                break;
+            case 'before':
+                result = diff < 0;
+                break;
+            case 'beforeOrOn':
+                result = diff <= 0;
+                break;
+            case 'between':
+                result = adapter.isBetween(itemValue, filterValue, filterValue2);
+                break;
+
+            case 'equalTo':
+            default:
+                result = adapter.dateTimesEqual(itemValue, filterValue);
+        }
+
+        return filter.exclude ? !result : result;
+    }
+
+    /**
+     * Boolean filtering strategy
+     * @param item data source array item.
+     * @param filter filter object
+     * @returns whether or not item should be included in data source array.
+     */
+    protected filterBoolean(item: T, filter: CollectionBooleanFilter): boolean {
+        const filterValue = filter.value;
+        const itemValue = this._getNestedValue(filter.field, item);
+        let result = false;
+
+        switch (filter.strategy) {
+            case 'equalTo':
+            default:
+                result = itemValue === filterValue;
+        }
+
+        return filter.exclude ? !result : result;
+    }
+
+    /**
+     * Array filtering strategy
+     * @param item data source array item.
+     * @param filter filter object
+     * @returns whether or not item should be included in data source array.
+     */
+    protected filterArray(item: T, filter: CollectionSelectFilter): boolean {
+        const filterValues = filter.value.map((v) => this.getSelectItemValue(v));
+        const itemValue = this._getNestedValue(filter.field, item);
+        let result = false;
+
+        switch (filter.strategy) {
+            case 'equalTo':
+            default:
+                result = filterValues.includes(itemValue);
+        }
+
+        return !filterValues.length || filter.exclude ? !result : result;
+    }
+
+    /**
+     * @hidden
+     */
+    protected _getNestedValue<P extends Record<string, any>>(key: string, object: P): any {
+        return key.split('.').reduce((a, b) => a[b], object);
+    }
```

Comment:
is this actually needed? I removed usages of this function in favour of lodash's `get` a while ago